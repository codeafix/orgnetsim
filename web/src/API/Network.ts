type Network = {
    links: Array<Link>
    nodes: Array<AgentState>;
    maxColors: number;
}

type Link = {
    source: string;
    target: string;
    strength: number;
    length: number;
}

type AgentState = {
    id: string;
    color: number;
    susceptability: number;
    influence: number;
    contrariness: number;
    change: number;
    type: string;
    fx: number;
    fy: number;
}
