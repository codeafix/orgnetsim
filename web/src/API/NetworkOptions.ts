type NetworkOptions = {
    linkTeamPeers: boolean;
    linkedTeamList: Array<string>;
    evangelistList: Array<string>;
    loneEvangelist: Array<string>;
    initColors: Array<number>;
    maxColors: number;
    agentsWithMemory: boolean;
}