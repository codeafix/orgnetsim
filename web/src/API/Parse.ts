type ParseOptions = {
    identifier: number;
    parent: number;
    regex: any;
    delimiter: string;
    payload: string;
}

export type { ParseOptions };