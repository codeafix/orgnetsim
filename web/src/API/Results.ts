type Results = {
    iterations: number;
    colors: Array<Array<number>>;
    conversations: Array<number>;
}

type ResultsCsv = {
    filename: string;
    blob: Blob;
}

export type { Results, ResultsCsv };