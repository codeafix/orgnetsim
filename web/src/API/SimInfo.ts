type SimInfo = {
    id: string;
    name: string;
    description: string;
    steps: Array<string>;
    options: NetworkOptions;
}