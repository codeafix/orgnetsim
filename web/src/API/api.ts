const API = {
    rootPath: "http://localhost:8080",
    emptySimList: {simulations:[],notes:""},
    sims: async function() {
        const response = await fetch(this.rootPath+"/api/simulation", {
        "method": "GET",
        "headers": {}
        }).catch(err => { console.log(err); 
        });
        return response.json();
    },
    get: async function(id){
        const response = await fetch(this.rootPath+"/api/simulation/"+id, {
            "method": "GET",
            "headers": {}
            }).catch(err => { console.log(err); 
        });
        return response.json();
    },
    getStep: async function(path){
        const response = await fetch(this.rootPath+path, {
            "method": "GET",
            "headers": {}
            }).catch(err => { console.log(err); 
        });
        return response.json();
    },
    getSteps: async function(sim){
        const response = await fetch(this.rootPath+"/api/simulation/"+sim.id+"/step", {
            "method": "GET",
            "headers": {}
            }).catch(err => { console.log(err); 
        });
        return response.json();
    },
    getResults: async function(sim){
        const response = await fetch(this.rootPath+"/api/simulation/"+sim.id+"/results", {
            "method": "GET",
            "headers": {}
            }).catch(err => { console.log(err); 
        });
        
        return response.json();
    },
    getResultsCsv: async function(sim){
        var fn = "results.csv";
        const data = await fetch(this.rootPath+"/api/simulation/"+sim.id+"/results", {
            "method": "GET",
            "headers": {
                "Content-Type": "text/csv"
            }
            }).then(response => {
                const contentDisp = response.headers.get('Content-Disposition');
                const regExpFilename = /filename="(?<filename>[^"]*)"/;
                fn = regExpFilename.exec(contentDisp)?.groups?.filename ?? fn;
                return response.blob();
            }).then(blob => {
                return {
                    filename: fn,
                    blob: blob,
                };
            }).catch(err => { console.log(err); 
            });
        return data;
    },
    update: async function(sim){
        const response = await fetch(this.rootPath+"/api/simulation/"+sim.id, {
            "method": "PUT",
            "headers": {
                'Content-Type': 'application/json'
            },
            "body": JSON.stringify(sim),
            })
            .catch(err => { console.log(err); 
            });
        return response.json();
    },
    updateStep: async function(step){
        const response = await fetch(this.rootPath+"/api/simulation/"+step.parent+"/step/"+step.id, {
            "method": "PUT",
            "headers": {
                'Content-Type': 'application/json'
            },
            "body": JSON.stringify(step),
            })
            .catch(err => { console.log(err); 
            });
        return response.json();
    },
    runsim: async function(sim, spec){
        const response = await fetch(this.rootPath+"/api/simulation/"+sim.id+"/run", {
            "method": "POST",
            "headers": {
                'Content-Type': 'application/json'
            },
            "body": JSON.stringify(spec),
            })
            .catch(err => { console.log(err); 
            });
            return response.json();
    },
    parse: async function(sim, pdata){
        const response = await fetch("http://localhost:8080/api/simulation/"+sim.id+"/parse", {
            "method": "POST",
            "headers": {
                'Content-Type': 'application/json'
            },
            "body": JSON.stringify(pdata),
            }).catch(err => { console.log(err); 
            });
            return response.json();
    },
    add: async function(name, description){
        var sim = {name:name,description:description};
        const response = await fetch(this.rootPath+"/api/simulation", {
            "method": "POST",
            "headers": {
                'Content-Type': 'application/json'
            },
            "body": JSON.stringify(sim),
            })
            .catch(err => { console.log(err); 
            });
        return response.json();
    },
    delete: async function(id){
        await fetch(this.rootPath+"/api/simulation/"+id, {
            "method": "DELETE",
            "headers": {}
            })
            .catch(err => { console.log(err); 
            });
    },
}

export default API