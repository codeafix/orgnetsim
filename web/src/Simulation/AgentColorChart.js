import React, { useEffect, useRef, useState } from 'react';
import { scaleLinear, max, select, stack, axisBottom, axisLeft, area } from 'd3';
import Color from './Color';
import API from '../api';
import Spinner from 'react-bootstrap/Spinner';

const AgentColorChart = (props) => {
    const chart = useRef(null);
    const [loading, setloading] = useState(false);

    const createChart = (results) => {
        if(!results['colors'][0]) return;
        const maxColors = props.sim.options['maxColors'];
        const margin = {top: 10, right: 60, bottom: 20, left: 40},
            vwidth = chart.current.parentElement.offsetWidth,
            cw = vwidth - margin.left - margin.right,
            vheight = Math.round(cw/1.6),
            ch = vheight - margin.top - margin.bottom;
        
        const resize = () => {
            const w = chart.current.parentElement.offsetWidth,
                h = Math.round((w -  margin.left - margin.right)/1.6);
            select(chart.current).attr('width', w)
                .attr('height', h);
        };

        select(window).on(
            'resize.' + select(chart.current.parentElement).attr('id'), 
            resize
        );
        
        const svg = select(chart.current);

        svg.attr('viewBox', `0 0 ${vwidth} ${vheight}`)
            .attr('preserveAspectRatio', 'xMinYMid')
            .call(resize);

        const dataMax = results['colors'][0].reduce((a, b) => a + b, 0);
        const iterations = results['iterations'];
        const chartData = results['colors'];

        var stackedData = stack()
            .keys(Color.colorValSlice(maxColors))
            .value(function(d, key){
                return d[key]
            })
            (chartData)
            
        //X Axis
        var xh = ch + margin.top;
        var xScale = scaleLinear()
            .domain([0, iterations])
            .range([0, cw]);
        svg.append("g")
            .attr("class", "small")
            .attr("transform", "translate(" + margin.left + "," + xh + ")")
            .call(axisBottom(xScale).ticks(10));
        
        //Add Y axis
        var yScale = scaleLinear()
            .domain([0, dataMax])
            .range([ch, 0]);
        svg.append("g")
            .attr("class", "small")
            .attr("transform", "translate(" + margin.left + "," + margin.top + ")")
            .call(axisLeft(yScale).ticks(10));
          
        svg.selectAll("mylayers")
            .data(stackedData)
            .enter()
            .append("path")
            .style("fill", (d) => Color.cssColorFromVal(d.key))
            .attr("transform", "translate(" + margin.left + "," + margin.top + ")")
            .attr("d", area()
                .x(function(d, i) { return xScale(i); })
                .y0(function(d) { return yScale(d[0]); })
                .y1(function(d) { return yScale(d[1]); })
            );
    }

    useEffect(() => {
        if (!props.sim['id']) {
            return
        }
        setloading(true);
        select(chart.current).selectAll("*").remove();
        API.getResults(props.sim).then(results => {
            createChart(results);
            setloading(false);
        }).catch(err => {
            console.error(err);
            setloading(false);
        });
    },[props.sim]);

    return <div>
            {loading && <Spinner animation="border" variant="info" />}
            <svg class="mb-3" ref={chart}/>
        </div>
}

export default AgentColorChart;