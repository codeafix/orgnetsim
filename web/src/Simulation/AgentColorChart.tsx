import React, { useEffect, useRef, useState } from 'react';
import { scaleLinear, select, stack, axisBottom, axisLeft, area } from 'd3';
import Color from './Color';
import API from '../API/api';
import Spinner from 'react-bootstrap/Spinner';
import { SimInfo } from '../API/SimInfo';
import { Results } from '../API/Results';

type AgentColorChartProps = {
    sim:SimInfo;
}

const AgentColorChart = (props:AgentColorChartProps) => {
    const chart = useRef<SVGSVGElement>(null);
    const [loading, setloading] = useState<boolean>(false);

    const createChart = (results:Results) => {
        if(!results['colors'][0]) return;
        if(!chart.current) return;
        var c = chart.current;
        if(!c.parentElement) return;

        const maxColors = props.sim.options['maxColors'];
        const margin = {top: 10, right: 60, bottom: 20, left: 40},
            vwidth = c.parentElement.offsetWidth,
            cw = vwidth - margin.left - margin.right,
            vheight = Math.round(cw/1.6),
            ch = vheight - margin.top - margin.bottom;
        
        const resize = () => {
            const c = chart.current;
            if(!c) return;
            if(!c.parentElement) return;
            const w = c.parentElement.offsetWidth,
                h = Math.round((w -  margin.left - margin.right)/1.6);
            select(c).attr('width', w)
                .attr('height', h);
        };

        select(window).on(
            'resize.' + select(c.parentElement).attr('id'), 
            resize
        );
        
        const svg = select(c);

        svg.attr('viewBox', `0 0 ${vwidth} ${vheight}`)
            .attr('preserveAspectRatio', 'xMinYMid')
            .call(resize);

        const dataMax = results['colors'][0].reduce((a, b) => a + b, 0);
        const iterations = results['iterations'];
        const chartData = results['colors'];

        var stackedData = stack<number[], number>()
            .keys(Color.colorValSlice(maxColors))(chartData);
            
        //X Axis
        var xh = ch + margin.top;
        var xScale = scaleLinear()
            .domain([0, iterations])
            .range([0, cw]);
        svg.append("g")
            .attr("class", "small")
            .attr("transform", "translate(" + margin.left + "," + xh + ")")
            .call(axisBottom(xScale).ticks(10));
        
        //Add Y axis
        var yScale = scaleLinear()
            .domain([0, dataMax])
            .range([ch, 0]);
        svg.append("g")
            .attr("class", "small")
            .attr("transform", "translate(" + margin.left + "," + margin.top + ")")
            .call(axisLeft(yScale).ticks(10));
          
        svg.selectAll("mylayers")
            .data(stackedData)
            .enter()
            .append("path")
            .style("fill", (d) => Color.cssColorFromVal(d.key))
            .attr("transform", "translate(" + margin.left + "," + margin.top + ")")
            .attr("d", area<number[]>()
                .x(function(d, i) { return xScale(i); })
                .y0(function(d) { return yScale(d[0]); })
                .y1(function(d) { return yScale(d[1]); })
            );
    }

    useEffect(() => {
        if (!props.sim['id']) {
            return
        }
        setloading(true);
        select(chart.current).selectAll("*").remove();
        API.getResults(props.sim).then(results => {
            createChart(results);
            setloading(false);
        }).catch(err => {
            console.error(err);
            setloading(false);
        });
    },[props.sim]);

    return <div id="chart-container">
            {loading && <Spinner animation="border" variant="info" />}
            <svg className="mb-3" ref={chart}/>
        </div>
}

export default AgentColorChart;