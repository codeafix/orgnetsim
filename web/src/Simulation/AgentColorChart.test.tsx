import React from 'react';
import {render} from '@testing-library/react';
import '../../node_modules/bootstrap/dist/css/bootstrap.css';
import '../index.css';
import { test, vi, expect } from 'vitest'
import { act } from 'react-dom/test-utils';
import AgentColorChart from './AgentColorChart';
import { SimInfo } from '../API/SimInfo';

const sim:SimInfo = {"id":"27f06fe2-6e82-44b0-af4a-6975d169ff48","name":"test","description":"","steps":["/api/simulation/27f06fe2-6e82-44b0-af4a-6975d169ff48/step/72e1e5cb-3f31-4afd-818f-2293076547f7","/api/simulation/27f06fe2-6e82-44b0-af4a-6975d169ff48/step/f62bd8e5-2027-4fca-9e02-92c6ae6468ac"],"options":{"linkTeamPeers":true,"linkedTeamList":[],"evangelistList":[],"loneEvangelist":[],"initColors":[0],"maxColors":2,"agentsWithMemory":false}};

vi.mock('../API/api');

test('renders without crashing', async () => {
    var result:any;
    await act(async () => {
        result = render(
            <AgentColorChart sim={sim}/>
        );
    });
    expect(result).toBeDefined();
    expect(result.asFragment()).toMatchSnapshot();
});
