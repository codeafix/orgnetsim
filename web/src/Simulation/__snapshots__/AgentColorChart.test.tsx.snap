// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    id="chart-container"
  >
    <svg
      class="mb-3"
      height="-62"
      preserveAspectRatio="xMinYMid"
      viewBox="0 0 0 -62"
      width="0"
    >
      <g
        class="small"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(40,-82)"
      >
        <path
          class="domain"
          d="M0.5,6V0.5H-99.5V6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-9.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-19.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            20
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-29.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            30
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-39.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            40
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-49.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            50
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-59.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            60
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-69.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            70
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-79.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            80
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-89.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            90
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-99.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            100
          </text>
        </g>
      </g>
      <g
        class="small"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
        transform="translate(40,10)"
      >
        <path
          class="domain"
          d="M-6,-91.5H0.5V0.5H-6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-91.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            0.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-83.83333333333333)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            0.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-76.16666666666667)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            1.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-68.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            1.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-60.83333333333334)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            2.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-53.16666666666666)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            2.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-45.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            3.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-37.83333333333333)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            3.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-30.16666666666667)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            4.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-22.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            4.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-14.83333333333333)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            5.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-7.16666666666667)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            5.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,0.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            6.0
          </text>
        </g>
      </g>
      <path
        d="M0,0L-1,-30.667L-2,0L-3,-15.333L-4,-30.667L-5,-30.667L-6,-30.667L-7,-15.333L-8,-15.333L-9,-15.333L-10,-15.333L-11,0L-12,-30.667L-13,-30.667L-14,-30.667L-15,-30.667L-16,-15.333L-17,-30.667L-18,-30.667L-19,-15.333L-20,-15.333L-21,0L-22,-15.333L-23,0L-24,0L-25,0L-26,0L-27,0L-28,-15.333L-29,-15.333L-30,-15.333L-31,0L-32,-30.667L-33,-15.333L-34,-30.667L-35,-15.333L-36,-15.333L-37,-15.333L-38,0L-39,-15.333L-40,-30.667L-41,-30.667L-42,0L-43,-15.333L-44,-15.333L-45,-15.333L-46,0L-47,-15.333L-48,-15.333L-49,-15.333L-50,-15.333L-51,-30.667L-52,-15.333L-53,-15.333L-54,-15.333L-55,-30.667L-56,-15.333L-57,0L-58,0L-59,0L-60,0L-61,-15.333L-62,-15.333L-63,-15.333L-64,-15.333L-65,0L-66,-30.667L-67,-15.333L-68,0L-69,-30.667L-70,-15.333L-71,-15.333L-72,-15.333L-73,-15.333L-74,-30.667L-75,-15.333L-76,-30.667L-77,-15.333L-78,-30.667L-79,-15.333L-80,0L-81,-15.333L-82,-15.333L-83,0L-84,0L-85,-15.333L-86,-15.333L-87,0L-88,-30.667L-89,-30.667L-90,-15.333L-91,-46L-92,-46L-93,-30.667L-94,-46L-95,-30.667L-96,-30.667L-97,-30.667L-98,0L-99,-30.667L-100,-30.667L-100,-92L-99,-92L-98,-92L-97,-92L-96,-92L-95,-92L-94,-92L-93,-92L-92,-92L-91,-92L-90,-92L-89,-92L-88,-92L-87,-92L-86,-92L-85,-92L-84,-92L-83,-92L-82,-92L-81,-92L-80,-92L-79,-92L-78,-92L-77,-92L-76,-92L-75,-92L-74,-92L-73,-92L-72,-92L-71,-92L-70,-92L-69,-92L-68,-92L-67,-92L-66,-92L-65,-92L-64,-92L-63,-92L-62,-92L-61,-92L-60,-92L-59,-92L-58,-92L-57,-92L-56,-92L-55,-92L-54,-92L-53,-92L-52,-92L-51,-92L-50,-92L-49,-92L-48,-92L-47,-92L-46,-92L-45,-92L-44,-92L-43,-92L-42,-92L-41,-92L-40,-92L-39,-92L-38,-92L-37,-92L-36,-92L-35,-92L-34,-92L-33,-92L-32,-92L-31,-92L-30,-92L-29,-92L-28,-92L-27,-92L-26,-92L-25,-92L-24,-92L-23,-92L-22,-92L-21,-92L-20,-92L-19,-92L-18,-92L-17,-92L-16,-92L-15,-92L-14,-92L-13,-92L-12,-92L-11,-92L-10,-92L-9,-92L-8,-92L-7,-92L-6,-92L-5,-92L-4,-92L-3,-92L-2,-92L-1,-92L0,-92Z"
        style="fill: SlateGray;"
        transform="translate(40,10)"
      />
      <path
        d="M0,0L-1,0L-2,0L-3,0L-4,0L-5,0L-6,0L-7,0L-8,0L-9,0L-10,0L-11,0L-12,0L-13,0L-14,0L-15,0L-16,0L-17,0L-18,0L-19,0L-20,0L-21,0L-22,0L-23,0L-24,0L-25,0L-26,0L-27,0L-28,0L-29,0L-30,0L-31,0L-32,0L-33,0L-34,0L-35,0L-36,0L-37,0L-38,0L-39,0L-40,0L-41,0L-42,0L-43,0L-44,0L-45,0L-46,0L-47,0L-48,0L-49,0L-50,0L-51,0L-52,0L-53,0L-54,0L-55,0L-56,0L-57,0L-58,0L-59,0L-60,0L-61,0L-62,0L-63,0L-64,0L-65,0L-66,0L-67,0L-68,0L-69,0L-70,0L-71,0L-72,0L-73,0L-74,0L-75,0L-76,0L-77,0L-78,0L-79,0L-80,0L-81,0L-82,0L-83,0L-84,0L-85,0L-86,0L-87,0L-88,0L-89,0L-90,0L-91,0L-92,0L-93,0L-94,0L-95,0L-96,0L-97,0L-98,0L-99,0L-100,0L-100,-30.667L-99,-30.667L-98,0L-97,-30.667L-96,-30.667L-95,-30.667L-94,-46L-93,-30.667L-92,-46L-91,-46L-90,-15.333L-89,-30.667L-88,-30.667L-87,0L-86,-15.333L-85,-15.333L-84,0L-83,0L-82,-15.333L-81,-15.333L-80,0L-79,-15.333L-78,-30.667L-77,-15.333L-76,-30.667L-75,-15.333L-74,-30.667L-73,-15.333L-72,-15.333L-71,-15.333L-70,-15.333L-69,-30.667L-68,0L-67,-15.333L-66,-30.667L-65,0L-64,-15.333L-63,-15.333L-62,-15.333L-61,-15.333L-60,0L-59,0L-58,0L-57,0L-56,-15.333L-55,-30.667L-54,-15.333L-53,-15.333L-52,-15.333L-51,-30.667L-50,-15.333L-49,-15.333L-48,-15.333L-47,-15.333L-46,0L-45,-15.333L-44,-15.333L-43,-15.333L-42,0L-41,-30.667L-40,-30.667L-39,-15.333L-38,0L-37,-15.333L-36,-15.333L-35,-15.333L-34,-30.667L-33,-15.333L-32,-30.667L-31,0L-30,-15.333L-29,-15.333L-28,-15.333L-27,0L-26,0L-25,0L-24,0L-23,0L-22,-15.333L-21,0L-20,-15.333L-19,-15.333L-18,-30.667L-17,-30.667L-16,-15.333L-15,-30.667L-14,-30.667L-13,-30.667L-12,-30.667L-11,0L-10,-15.333L-9,-15.333L-8,-15.333L-7,-15.333L-6,-30.667L-5,-30.667L-4,-30.667L-3,-15.333L-2,0L-1,-30.667L0,0Z"
        style="fill: Blue;"
        transform="translate(40,10)"
      />
    </svg>
  </div>
</DocumentFragment>
`;
